browser/src/react-redux/App/SignupPage.jsx

```diff
@@ -0,0 +1,156 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import setUser from './setUser.action'
+import axios from 'axios';
+
+const initialState = {
+  confirmPassword: '',
+  email: '',
+  password: '',
+  emailError: '',
+  generalError: '',
+  passwordError: ''
+};
+
+const mapDispatchToProps = dispatch => {
+  return {
+    setUser: user => {
+      dispatch(setUser(user))
+    }
+  }
+}
+
+class SignupPage extends Component {
+  static propTypes = {
+    match: PropTypes.object.isRequired,
+    location: PropTypes.object.isRequired,
+    history: PropTypes.object.isRequired,
+    setUser: PropTypes.func.isRequired
+  };
+
+  state = initialState
+
+  handleChange = (event) => {
+    this.setState({
+      [event.target.name]: event.target.value
+    });
+  }
+
+  getConfirmPasswordError = () => {
+    return (this.state.confirmPassword && this.state.password !== this.state.confirmPassword)
+      ? 'Passwords do not match.'
+      : null;
+  }
+
+  enableButton = () => {
+    return (
+      this.state.confirmPassword &&
+      this.state.password &&
+      this.state.confirmPassword &&
+      this.state.password === this.state.confirmPassword
+    )
+      ? false
+      : true;
+  }
+
+  onSubmit = (e) => {
+    this.setState(initialState);
+
+    e.preventDefault();
+
+    const user = {
+      email: this.state.email,
+      password: this.state.password
+    };
+
+    axios.post('/api/user', user)
+      .then(res => {
+        if (res.request.status === 201) {
+          //set current user with redux store.
+          this.props.setUser({ email: user.email });
```

Comment:
We will only ever get a 201 with create, otherwise it's an error. So that we're not overcomplicating this code. Let's get rid of the status check and execute the logic right away.